zulip_bots/zulip_bots/bots/chess/test_chess.py

```diff
@@ -0,0 +1,168 @@
+#!/usr/bin/env python
+
+from zulip_bots.test_lib import BotTestCase
+
+class TestChessBot(BotTestCase):
+    bot_name = "chess"
+
+    START_RESPONSE = '''New game! The board looks like this:
+
+```
+  a b c d e f g h
+8 ♜ ♞ ♝ ♛ ♚ ♝ ♞ ♜ 8
+7 ♟ ♟ ♟ ♟ ♟ ♟ ♟ ♟ 7
+6 · · · · · · · · 6
+5 · · · · · · · · 5
+4 · · · · · · · · 4
+3 · · · · · · · · 3
+2 ♙ ♙ ♙ ♙ ♙ ♙ ♙ ♙ 2
+1 ♖ ♘ ♗ ♕ ♔ ♗ ♘ ♖ 1
+  a b c d e f g h
+```
+
+
+Now it's **white**'s turn.
+
+
+To make your next move, respond to Chess Bot with
+
+```rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR w KQkq - 0 1: do <your move>```
```

Comment:
I feel like it would be easier to communicate this using examples.  Chess players are more likely to know chess notation than being able to parse out what you're saying here, I think.

Reply:
OK, I'll add an example. Thanks! The reason the beginning part (`rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR w KQkq - 0 1`) is needed is because the bot can't carry over data from earlier messages, so instead the board information has to be sent along with the move.

Reply:
Hey @skunkmb! We do support saving the data. I am in a bit of a hurry so cannot link to exact code samples but take a look at the `virtual_fs` bot and try running it once as well.

We do not need to send the data with each move and can use our bot handler storage. :smiley:

Reply:
@showell @aero31aero I added to my commit, and now the match data is saved between responses. I didn't know you could do that 😃 ! Thanks! 

I think this makes it a lot more intuitive for the user to understand what they're supposed to do. Now the user doesn't have to have all of the match data at the front of their response.